rust/lance-core/src/utils/mask.rs

```diff
@@ -561,6 +709,62 @@ mod tests {
         assert!(combined.selected(1));
     }
 
+    #[test]
+    fn test_map_insert_range() {
+        let ranges = &[
+            (0..10),
+            (40..500),
+            ((u32::MAX as u64 - 10)..(u32::MAX as u64 + 20)),
+        ];
+
+        for range in ranges {
+            let mut mask = RowIdTreeMap::default();
+
+            let count = mask.insert_range(range.clone());
+            let expected = range.end - range.start;
+            assert_eq!(count, expected);
+
+            let count = mask.insert_range(range.clone());
+            assert_eq!(count, 0);
+
+            let new_range = range.start + 5..range.end + 5;
+            let count = mask.insert_range(new_range.clone());
+            assert_eq!(count, 5);
+        }
+
+        let mut mask = RowIdTreeMap::default();
+        let count = mask.insert_range(..10);
+        assert_eq!(count, 10);
+        assert!(mask.contains(0));
+
+        let count = mask.insert_range(20..=24);
+        assert_eq!(count, 5);
+
+        mask.insert_fragment(0);
+        let count = mask.insert_range(100..200);
+        assert_eq!(count, 0);
+    }
+
+    #[test]
+    fn test_map_remove() {
+        let mut mask = RowIdTreeMap::default();
+
+        assert!(!mask.remove(20));
+
+        mask.insert(20);
+        assert!(mask.contains(20));
+        assert!(mask.remove(20));
+        assert!(!mask.contains(20));
+
+        mask.insert_range(10..=20);
+        assert!(mask.contains(15));
+        assert!(mask.remove(15));
+        assert!(!mask.contains(15));
+
+        // We don't test removing from a full fragment, because that would take
+        // a lot of memory.
```

Comment:
I've been making large tests and just marking them `ignore`.  It's helpful during refactoring to manually make sure I didn't botch anything but I don't know how maintainable it is.